EIPS/eip-7645.md

```diff
@@ -0,0 +1,73 @@
+---
+eip: 7645
+title: Alias ORIGIN to SENDER
+description: Eliminate ORIGIN tech debt to lay groundwork for account abstraction and close security holes
+author: Cyrus Adkisson (@cyrusadkisson), Eirik Ulversøy (@EirikUlversoy)
+discussions-to: https://ethereum-magicians.org/t/eip-7645-alias-origin-to-sender/19047
+status: Draft
+type: Standards Track
+category: Core
+created: 2024-03-03
+---
+
+## Abstract
+
+This EIP proposes aliasing the ORIGIN opcode to the SENDER opcode within the Ethereum Virtual Machine (EVM). The purpose of this change is to move Ethereum closer to enabling account abstraction by harmonizing the treatment of externally owned accounts (EOAs) and smart contracts and to address the security concerns associated with the use of ORIGIN that have and will continue to surface in all or most account abstraction proposals. 
+
+## Motivation
+
+The ORIGIN opcode in Ethereum returns the address of the account that started the transaction chain, differing from the SENDER (or CALLER) opcode, which returns the address of the direct caller. The use of ORIGIN has been discouraged and deemed deprecated since mid-2016 due to the security problems it introduces, such as susceptibility to phishing attacks and other vulnerabilities where the distinction between the original sender and the immediate sender can be exploited.
+
+For instance, if an [ERC-4337](./eip-4337.md) bundler has tokens or other authority in a smart contract determined by ORIGIN, any of the transactions it bundles can hijack this authority since ORIGIN remains the bundler address throughout each child transaction.
+
+More apropos in the current context of EVM evolution, the differentiation between the ORIGIN and SENDER opcodes presents a challenge for all account abstraction efforts, such as those outlined in [EIP-7377](./eip-7377.md) and [EIP-3074](./eip-3074.md), because any move towards account abstraction must address the ORIGIN opcode's role, either by modifying or completely bypassing it. Without addressing this, the ORIGIN opcode stands as a barrier to the evolution of Ethereum's account model towards greater flexibility and functionality.
```

Comment:
Just a heads up, linking to non-final proposals will prevent yours from advancing.

Reply:
I'll remove the references to 7377 and 3074, but why is EIP-4337 still "draft" if it's in use today?

Reply:
That's up to the authors of ERC-4337.